src/index.ts

```diff
@@ -1,117 +1,203 @@
+import {globalExternals} from '@fal-works/esbuild-plugin-global-externals'
+import type {ModuleInfo} from '@fal-works/esbuild-plugin-global-externals'
+
+type ESBuildOptions = BuildOptions & {write: false}
+
+type BundleMDXOptions = {
+  /**
+   * The dependencies of the MDX code to be bundled
+   *
+   * @example
+   * ```
+   * bundleMDX(mdxString, {
+   *   files: {
+   *     './components.tsx': `
+   *       import * as React from 'react'
+   *
+   *      type CounterProps = {initialCount: number, step: number}
+   *
+   *       function Counter({initialCount = 0, step = 1}: CounterProps) {
+   *         const [count, setCount] = React.useState(initialCount)
+   *         const increment = () => setCount(c => c + step)
+   *         return <button onClick={increment}>{count}</button>
+   *       }
+   *     `
+   *   },
+   * })
+   * ```
+   */
+  files?: Record<string, string>
+  /**
+   * The remark plugins you want applied when compiling the MDX
+   *
+   * NOTE: Specifying this will override the default value for stripping
+   * frontmatter remark-frontmatter
+   */
+  remarkPlugins?: Array<unknown>
+  /**
+   * This allows you to modify the built-in esbuild configuration. This can be
+   * especially helpful for specifying the compilation target.
+   *
+   * @example
+   * ```
+   * bundleMDX(mdxString, {
+   *   esbuildOptions(options) {
+   *     options.target = [
+   *       'es2020',
+   *       'chrome58',
+   *       'firefox57',
+   *       'safari11',
+   *       'edge16',
+   *       'node12',
+   *     ]
+   *     return options
+   *   }
+   * })
+   * ```
+   */
+  esbuildOptions?: (options: ESBuildOptions) => ESBuildOptions
+  /**
+   * Any variables you want treated as global variables in the bundling.
+   *
+   * NOTE: These do not have to be technically global as you will be providing
+   * their values when you use getMDXComponent, but as far as esbuild is concerned
+   * it will treat these values as global variables so they will not be included
+   * in the bundle.
+   *
+   * @example
+   * ```
+   * bundlMDX(mdxString, {
+   *   globals: {'left-pad': 'myLeftPad'},
+   * })
+   *
+   * // then later
+   *
+   * import leftPad from 'left-pad'
+   *
+   * const Component = getMDXComponent(result.code, {myLeftPad: leftPad})
+   * ```
+   */
+  globals?: Record<string, string | ModuleInfo>
+}
 
 async function bundleMDX(
   mdxSource: string,
   {
     files = {},
-    remarkPlugins = [],
-    rollup: {
-      getInputOptions = (options: RollupOptions) => options,
-      getOutputOptions = (options: OutputOptions) => options,
-    } = {},
-  }: {
-    files?: Record<string, string>
-    remarkPlugins?: Array<unknown>
-    rollup?: {
-      getInputOptions?: (options: RollupOptions) => RollupOptions
-      getOutputOptions?: (options: OutputOptions) => OutputOptions
-    }
-  } = {},
+    esbuildOptions = (options: ESBuildOptions) => options,
+    remarkPlugins = [remarkFrontmatter],
```

Comment:
I was unable to get `remark-mdx-frontmatter` to work with `@mdx-js/mdx` :-(

Reply:
> I was unable to get `remark-mdx-frontmatter` to work with `@mdx-js/mdx` :-(

Yup, it’s working on the new AST which isn’t available there yet